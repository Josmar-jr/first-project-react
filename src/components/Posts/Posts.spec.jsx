import { render, screen } from "@testing-library/react"
import { Posts } from '.'

const props = {
  posts: [
    {
      id: 1,
      title: 'Title 1',
      body: 'Body 1',
      cover: 'file.png'
    },
    {
      id: 2,
      title: 'Title 2',
      body: 'Body 2',
      cover: 'file.png'
    },
    {
      id: 3,
      title: 'Title 3',
      body: 'Body 3',
      cover: 'file.png'
    }
  ]
}

describe('<Posts />', () => {
  it('should render Posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /Title/i }))
      .toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /Title/i }))
      .toHaveLength(3);
    expect(screen.getAllByText(/body/i))
      .toHaveLength(3);
      expect(screen.getByRole('img', { name: /Title 3/i }))
      .toHaveAttribute('src', 'file.png');
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  })
});
