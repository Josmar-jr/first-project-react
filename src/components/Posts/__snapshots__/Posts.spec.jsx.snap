// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<section
  class="sc-gtsrHT fOfBmW"
>
  <div
    class="sc-bdnxRM kSBbAX"
  >
    <img
      alt="Title 1"
      src="file.png"
    />
    <h2>
      Title 1
    </h2>
    <p>
      Body 1
    </p>
  </div>
  <div
    class="sc-bdnxRM kSBbAX"
  >
    <img
      alt="Title 2"
      src="file.png"
    />
    <h2>
      Title 2
    </h2>
    <p>
      Body 2
    </p>
  </div>
  <div
    class="sc-bdnxRM kSBbAX"
  >
    <img
      alt="Title 3"
      src="file.png"
    />
    <h2>
      Title 3
    </h2>
    <p>
      Body 3
    </p>
  </div>
</section>
`;
